pkg/dataobj/querier/iter.go

```diff
@@ -328,10 +339,12 @@ func newSampleIterator(ctx context.Context,
 				// TODO(twhitney): when iterating over multiple extractors, we need a way to pre-process as much of the line as possible
 				// In the case of multi-variant expressions, the only difference between the multiple extractors should be the final value, with all
 				// other filters and processing already done.
+				statistics.AddDecompressedLines(1)
 				value, parsedLabels, ok := streamExtractor.Process(timestamp, record.Line, record.Metadata...)
 				if !ok {
 					continue
 				}
+				statistics.AddPostFilterLines(1)
```

Comment:
this unfortunately doesn't account for any filtering done by the dataset reader, but I think that's fine for now (as long as we remember that's how it works 😛)

Reply:
Yes, I'm thinking to add more comprehensive stats but only show a small number of them for benchmarks here. The existing chunk stats don't really cover the new reading approach, where we read and discard rows before even getting to the pipeline, so I'll have to think about that a bit more carefully in a follow up PR. This should get us started though.